<launch>

    <param name="/use_sim_time" type="bool" value="false"/>

    <rosparam command="load" file="$(find rnw_ros)/config/swarm_rnw_config.yaml"/>

    <node pkg="mocap_optitrack" type="mocap_node" name="mocap_node" respawn="false" launch-prefix="" required="true" output="log">
        <rosparam file="$(find mocap_optitrack)/config/mocap.yaml" command="load" />
    </node>

    <node pkg="mocap2odom" type="mocap2odom" name="mocap2odom" output="screen" >
        <param name="publish_uav_odom" type="bool" value="true"/>
        <rosparam file="$(find mocap2odom)/config/calib.yaml" command="load" />
    </node>

    <node pkg="rnw_ros" type="pub_cone_state_node" name="pub_cone_state_node" required="true" output="screen">
        <param name="cut_euler_velocity" type="bool" value="true"/>
        <param name="max_euler_velocity" value="10"/>
    </node>

    <node pkg="rnw_ros" type="odom2tf_node" name="drone1_tf" output="screen">
        <remap from="~odom" to="/drone1/odom"/>
        <param name="name" value="odom_drone1" />
    </node>

    <node pkg="rnw_ros" type="odom2tf_node" name="drone2_tf" output="screen">
        <remap from="~odom" to="/drone2/odom"/>
        <param name="name" value="odom_drone2" />
    </node>

    <node name="rviz_traj" pkg="rviz" type="rviz" output="log" args="-d $(find rnw_ros)/config/swarm.rviz" />

    <!--    <node pkg="rqt_multiplot" type="rqt_multiplot" name="rqt_multiplot"-->
    <!--          args="&#45;&#45;multiplot-config $(find rnw_ros)/config/rqt_multiplot_vicon_imu.xml &#45;&#45;multiplot-run-all"/>-->

<!--    <node pkg="rnw_ros" type="visualize_traj_node" name="traj_viz_drone1" output="screen" required="true">-->
<!--        <param name="lift_dt" type="double" value="0.05"/>-->
<!--        <param name="length_g" type="double" value="0.5"/>-->
<!--        <param name="clear_after_n_sec" type="double" value="0"/>-->
<!--        <remap from="~poly_traj" to="/drone1/traj"/>-->
<!--    </node>-->

<!--    <node pkg="rnw_ros" type="visualize_traj_node" name="traj_viz_drone2" output="screen" required="true">-->
<!--        <param name="lift_dt" type="double" value="0.05"/>-->
<!--        <param name="length_g" type="double" value="0.5"/>-->
<!--        <param name="clear_after_n_sec" type="double" value="0"/>-->
<!--        <remap from="~poly_traj" to="/drone2/traj"/>-->
<!--    </node>-->

<!--    <node pkg="rnw_ros" type="visualize_cone_node" name="visualize_cone_node" output="screen" required="true">-->
<!--        <param name="cone_color_r" type="double" value="0"/>-->
<!--        <param name="cone_color_g" type="double" value="1"/>-->
<!--        <param name="cone_color_b" type="double" value="0"/>-->
<!--    </node>-->

    <node name="record" pkg="rosbag" type="record" output="screen" args="-a" />

</launch>
